packages/smooth_app/lib/functions/launchURL.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:flutter/material.dart';
+import 'package:url_launcher/url_launcher.dart';
+import 'package:basic_utils/basic_utils.dart';
+
+
+class Launcher{
+
+  ///
+  /// ifOFF (is Open food facts)
+  /// when true add language identifier at start
+  ///
+  /// !!!
+  /// NO  https://de.openfoodfacts.org/...
+  /// YES https://   openfoodfacts.org/...
+  /// !!!
+  ///
+
+
+
+  Future<void> launchURL(BuildContext context, String url, bool isOFF ) async{
+
+    String openURL;
+    String localeString;
+
+    if(isOFF){
+        final Locale locale = Localizations.localeOf(context);
+        if(locale.countryCode.toString() == null){
+          localeString = 'world.';
+        }
+        else{
+          localeString = '${locale.countryCode.toString()}.';
+        }
+        openURL = StringUtils.addCharAtPosition(url, '$localeString', 8);
```

Comment:
Not very safe, assumes that the url contains the 'https://' prefix, maybe `url.replace('openfoodfacts.org', 'S{localeString}openfoodfacts.org');` is better. Also we should check if the url doesn't already contain a local.

Reply:
I have improved. Now it should work better. I just couldn't think of what to do when there is already a country code. Now there's been a bug in doing so.
 I hope future contributors will test their links.